### ISSUE TITLE
Does "--esnext" support static class properties?

### ISSUE THREAD

**Issue by moroshko** [jscs-dev/node-jscs#1344]
I use static class properties in my code:

    class MyComponent extends React.Component {
      static propTypes = {
        expanded: false
      };
      ...
    }

I use **--esnext**, but still get an error:

    parseError: Unexpected token =

1. Is there a way to ignore this error?
2. Are there any plans to support static class properties?
3. Where could I find all the es6 features that **--esnext** supports?

**Comment by hzoo** [comment]
You can use esprima-fb as the parser if it supports that syntax.

**Comment by moroshko** [comment]
Does it make sense to use both `--esnext --esprima=esprima-fb`, or I should be using only one of these?

**Comment by hzoo** [comment]
Yes yo

**Comment by markelog** [comment]
I'm not sure who `esprima-fb` is relevant here, it seems Esprima nor in 2.2 nor in harmony branch does not support static properties.

Re-opining for further invistigation, @mikesherov @ariya please advice

**Comment by hzoo** [comment]
Ok wasn't sure if it supported it so I gave the suggestion: `if it supports that syntax`. But yeah it seems that it's not - https://github.com/facebook/esprima/issues/63.

**Comment by ariya** [comment]
@markelog  Still not supported, if you paste the sample code to http://esprima.org/demo/parse.html.

**Comment by markelog** [comment]
Okay, upon further investigation it is revealed that this is not valid es6, it's not even in the list of proposals for es7, i guess babel only supports it because of <a href="https://github.com/rwaldron/tc39-notes/blob/master/es6/2015-03/mar-24.md#61-es6-class-properties">this</a> discussion, which seems premature even for babel.

/cc @sebmck

**Comment by sebmck** [comment]
@markelog It's based on [this](https://gist.github.com/jeffmo/054df782c05639da2adb) proposal which was proposed at the last TC39.

**Comment by markelog** [comment]
Yeah, i have a link to this gist above, isn't to early? Since this proposal, even if submitted (which is currently <a href="https://github.com/tc39/ecma262">not</a>) will likely change?

**Comment by sebmck** [comment]
@markelog It's behind a flag, you have to explicitly turn on support for stage 0 features. It's in Babel because it's meant to provide feedback to the committee and improve the proposal. ie. to find out what does and doesn't work. If the proposal changes then Babel changes with it. Decorators were implemented in Babel when it was still stage 0 (now stage 1) and have provided critical feedback to their design and use and it's influenced the proposal/spec.
